code/modules/xenoarchaeology/xenoartifact_labeler.dm

```diff
@@ -0,0 +1,247 @@
+
+// Not to be confused with labeler
+/obj/item/xenoartifact_label
+	icon = 'icons/obj/xenoarchaeology/xenoartifact_sticker.dmi'
+	icon_state = "sticker_star"
+	name = "artifact label"
+	desc = "An adhesive label describing the characteristics of a Xenoartifact."
+	var/info = "" 
+	var/set_name = FALSE
+	var/mutable_appearance/sticker_overlay
+	var/list/trait_list = list() //List of traits used to compare and generate modifier.
+	var/obj/item/xenoartifact/xenoa_target
+	///reference for timer
+	var/fall_timer
+
+/obj/item/xenoartifact_label/Initialize()
+	icon_state = "sticker_[pick("star", "box", "tri", "round")]"
+	var/sticker_state = "[icon_state]_small"
+	sticker_overlay = mutable_appearance(icon, sticker_state)
+	sticker_overlay.layer = FLOAT_LAYER
+	sticker_overlay.appearance_flags = RESET_ALPHA
+	sticker_overlay.appearance_flags = RESET_COLOR
+	..()
+	
+/obj/item/xenoartifact_label/afterattack(atom/target, mob/user, instant = FALSE)
+	if(istype(target, /mob/living))
+		to_chat(target, "<span class='warning'>[user] attempts to stick a [src] to you!</span>")
+		to_chat(user, "<span class='warning'>You attempt to stick a [src] on [target]!</span>")
+		if(do_after(user, 30, target = target))
+			if(!user.temporarilyRemoveItemFromInventory(src))
+				return
+		else
+			return
+		add_sticker(target)
+		fall_timer = addtimer(CALLBACK(src, .proc/remove_sticker, target), 15 SECONDS, TIMER_STOPPABLE)
+		return TRUE
+	else if(istype(target, /obj/item/xenoartifact))
+		xenoa_target = target
+		if(set_name) //You can update the name now
+			xenoa_target.name = name
+		calculate_modifier(xenoa_target)
+		add_sticker(xenoa_target)
+		if(set_name)
+			xenoa_target.name = name
+		if(info)
+			var/textinfo = list2text(info)
+			xenoa_target.label_desc = "There's a sticker attached, it says-\n[textinfo]"
+		return TRUE
+	
+/obj/item/xenoartifact_label/proc/add_sticker(mob/target)
+	if(locate(/obj/item/xenoartifact_label) in target) //Remove old stickers
+		qdel(locate(/obj/item/xenoartifact_label) in target)
+	target.add_overlay(sticker_overlay)
+	forceMove(target)
+
+/obj/item/xenoartifact_label/proc/remove_sticker(mob/target) //Peels off
+	target.cut_overlay(sticker_overlay)
+	forceMove(get_turf(target))
+
+/obj/item/xenoartifact_label/proc/calculate_modifier(obj/item/xenoartifact/X) //Modifier based off preformance of slueth. To:Do revisit this, complexity would be nice
+	var/datum/xenoartifact_trait/trait
+	var/datum/component/xenoartifact_pricing/xenop = X.GetComponent(/datum/component/xenoartifact_pricing)
+	if(!xenop)
+		return
+	xenop.modifier = initial(xenop.modifier)
+	for(var/t in trait_list)
+		trait = new t
+		if(X.get_trait(trait))
+			xenop.modifier += 0.15 
+		else
+			xenop.modifier -= 0.35
+
+/obj/item/xenoartifact_label/proc/list2text(list/listo) //list2params acting weird. Probably already a function for this.
+	var/text = ""
+	for(var/X in listo)
+		if(X)
+			text = "[text] [X]\n"
```

Comment:
If you have to check for listo containing null values, then something is messed up somewhere else and this seems like a sign of poor code quality elsewhere.

Reply:
No real reason to check it, I think I convinced myself I needed to do extensive checking for everything.